### ISSUE TITLE
PPU: mm/bmm select the _hygon kernel, whose num_ldmatrixes kwarg PPU triton rejects

### ISSUE THREAD

**Issue by lvyufeng** [FlagOpen/FlagGems#6225]
## Summary

On PPU, `flag_gems`' `mm` and `bmm` route into the `_hygon` mm kernel, which
passes a `num_ldmatrixes` kwarg to triton's `mm_kernel`. PPU's triton build does
not accept that keyword. `mm` raises `KeyError` at
`triton/runtime/jit.py:_pack_args`; `bmm` reaches the same kernel and stalls
inside triton compilation instead of raising.

The kernel selection appears to be vendor-baked rather than resolved against the
active triton backend, so a PPU process picks up a Hygon-specific kernel.

## Environment

- Platform: PPU (CUDA-ABI compatible accelerator, `torch.cuda.is_available()` is
  True and reports real PPU devices)
- torch: 2.10.0+cpu with a vendor `libtorch_cuda.so` bundled by the PPU SDK
- triton: PPU vendor build, copied into the venv from the PPU SDK
- FlagGems: routed through FlagOS Torch-FL's FlagGems Python path
- Python 3.12

## Reproduction

Route `mm` to the FlagGems Python path on PPU and call it:

```python
import torch_fl, torch
a = torch.randn(64, 128, device="flagos:0")
b = torch.randn(128, 32, device="flagos:0")
torch.mm(a, b)     # KeyError in triton/runtime/jit.py:_pack_args
torch.bmm(...)     # same kernel; stalls in triton compilation
```

## Observed behavior

Measured on a PPU CI runner. The op is functionally correct in the end, so this
shows up as latency rather than a failure — which is what made it hard to spot:

```
tests/integration/test_factory_ops.py    46 passed in 1801.85s

  TestArithmetic::test_div    PASSED  22:25:38
  TestMatrixOps::test_mm      PASSED  22:53:48   <- 28 minutes in one 64x128 mm
  TestMatrixOps::test_matmul  PASSED  22:53:48   <- instant, cache now warm
  TestMatrixOps::test_bmm     PASSED  22:55:36   <- 108s
```

The same file with `mm`/`bmm` on the vendor kernel: **46 passed in 2.87s**.

A Qwen3-0.6B inference test in the same run passed all 4 cases (99.30s) and then
terminated with **SIGSEGV at interpreter exit** (exit code -11). I have not
isolated the segfault to a specific FlagGems call, so I am reporting it only as
correlated: it appears when `mm`/`bmm` are routed to FlagGems on PPU and not when
they are on the vendor kernel.

Every assertion passed in both configurations. Only the wall time and the exit
status differ.

## Expected behavior

`mm`/`bmm` on a non-Hygon backend should not select the `_hygon` kernel, and a
kwarg the active triton does not accept should raise immediately rather than
after a ~28 minute compile.

## Other backends

Nine other FlagGems ops (`abs`, `add`, `embedding`, `mean`, `neg`, `silu`,
`softmax`, `sum`, `where`) pass on PPU in 73.88s in the same run, so the
FlagGems Python path itself works there. This is specific to the mm/bmm kernel
selection.

This may share a root cause with the vendor-baked kernel qualnames I have seen
elsewhere: generated kernels naming `_hygon.ops.*` are not present on other
platforms.

## Workaround

Pinning `mm` and `bmm` to the vendor kernel for PPU in our routing config. The
other FlagGems routes are kept. Happy to test a fix on PPU hardware.
